tests/scancode/test_cli.py

```diff
@@ -452,3 +464,41 @@ def test_scan_can_run_from_other_directory():
         print(stderr)
     assert rc == 0
     check_scan(test_env.get_test_loc(expected_file), result_file, strip_dates=True, regen=False)
+
+
+class TestShortenFilename(TestCase):
```

Comment:
You should add tests that test at the boundaries: e.g. when you have exactly the max length that will be shorted., then that minus one and that plus one. This way this would spec your function fully. You may have them already, but if this is the case, the test function names are not explicit enough